File: domain/src/main/java/realm/DataManager.java
/**
 * Created by sayler on 2016-12-10.
 * <p>
 * Copyright 2016 MiQUiDO <http://www.miquido.com/>. All rights reserved.
 */
package realm;

public interface DataManager<T> {
  void setSourceFile(String sourceFilePath);

  boolean isOpen();

  void close();

  T getDao();
}

File: domain/src/main/java/com/sayler/gina/domain/DataManager.kt
/**
 * Created by sayler on 2016-12-10.
 *
 *

 */
package com.sayler.gina.domain

interface DataManager<T> {
    fun setSourceFile(sourceFilePath: String)

    val isOpen: Boolean

    fun getSourceFilePath(): String

    fun close()

    val dao: T
}
